checkpoint2/Program.cs

```diff
@@ -0,0 +1,348 @@
+                input = Console.ReadLine();
+                if (input.ToLower() == "c")
+                {
+                    CallItADay();
+                    return;
+                }
+                if(!Int32.TryParse(input, out fromX))
+                {
+                    System.Console.WriteLine("Your input is invalid. Only input numbers 0-7. Try again.");
+                    return;
+                }
+                
+                Console.WriteLine("Enter Pickup Column: ");
+                input = Console.ReadLine();
+                if(!Int32.TryParse(input, out fromY))
+                {
+                    System.Console.WriteLine("Your input is invalid. Only input numbers 0-7. Try again.");
+                    return;
+                }
+                CheckPlayerInputFrom();
+            }
+            void CheckPlayerInputFrom(){
+                //INPUT CHECKS / FROM
+                Checker checkerToMove = SelectChecker(fromX, fromY);   
+                if (!validInput.Contains(fromX) || !validInput.Contains(fromY)) // CHECKS IF INPUT IS OUT OF BOUNDS
+                {
+                    Console.WriteLine("Your selection is out of bounds of the board");
+                    return;
+                }       
+                else if (checkerToMove == null) // CHECKS IF A CHECKER IS THERE
+                {
+                    Console.WriteLine("You chose an empty space. Please choose again.");
+                    return;
+                }
+                else if (checkerToMove.color != playerColor) //CHECKS TO SEE IF IT'S THE RIGHT PLAYER's CHECKER
+                {
+                    Console.WriteLine("You chose the wrong colored checker. Please choose again.");
+                    return;
+                }
+                GetPlayerInputTo();
+            }
+            void GetPlayerInputTo(){
+                //GETTING PLAYER INPUT / TO
+                Console.WriteLine("Enter Placement Row: ");
+                input = Console.ReadLine();
+                if(!Int32.TryParse(input, out toX))
+                {
+                    System.Console.WriteLine("Your input is invalid. Only input numbers 0-7. Try again.");
+                    return;
+                }
+                Console.WriteLine("Enter Placement Column: ");
+                input = Console.ReadLine();
+                if(!Int32.TryParse(input, out toY))
+                {
+                    System.Console.WriteLine("Your input is invalid. Only input numbers 0-7. Try again.");
+                    return;
+                }
+                CheckPlayerMove();
+            }
+            void CheckPlayerMove(){
+                //MOVEMENT CHECKS / TO
+                Checker checkerToMove = SelectChecker(fromX, fromY);   
+                if (!validInput.Contains(toX) || !validInput.Contains(toY)) // CHECKS IF DESTINATION IS OUT OF BOUNDS
```

Comment:
you call SelectChecker(toX, toY) multiple times, could probably have made a targetChecker variable to hold that and used it. 
``` Checker targetChecker = SelectChecker(toX,toY);```

Seems like you are relying on the grid[toX,toY] to tell if the destination is empty square.
You could use the targetChecker to check, and that way the grid is just for drawing
``` bool emptyTargetSpace = targetChecker == null;
``` bool isEnemyTarget = targetChecker == null ? false : targetChecker.color != checkerToMove.color;```